### PR TITLE
add configurable discoveryBurst and discoveryQPS

### PR DESCRIPTION
**Pull request by roytman**
Allows to configure discovery K8s client parameters for helm operations.
/closes 1740

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

### CODE REVIEW

File: pkg/environment/environment.go [hunk 1]
```diff
@@ -179,6 +182,29 @@ func GetHelmWaitTimeout() (time.Duration, error) {
 	return time.Duration(timeout) * time.Second, nil
 }
 
+// GetDiscoveryBurst returns the K8s discovery burst value if it is set, otherwise it returns -1
+func GetDiscoveryBurst() (int, error) {
+	burstStr := os.Getenv(DiscoveryBurst)
+	if burstStr == "" {
+		return -1, nil
+	}
+	return strconv.Atoi(burstStr)
```
Comment (shlomitk1) [inline pkg/environment/environment.go:191]: In case of an error, 0 is returned as a value.

**Review by revit13 — APPROVED** [review]

File: pkg/helm/helm.go [hunk 2]
```diff
@@ -299,6 +308,13 @@ func (r *Impl) GetConfig(kubeNamespace string, log action.DebugLog) (*action.Con
 	config := &genericclioptions.ConfigFlags{
 		Namespace: &kubeNamespace,
 	}
+	if r.discoveryBurst != -1 {
```
Comment (shlomitk1) [inline pkg/helm/helm.go:311]: I think it's better to refer to positive values only: **> 0**

File: pkg/helm/helm.go [hunk 3]
```diff
@@ -299,6 +308,13 @@ func (r *Impl) GetConfig(kubeNamespace string, log action.DebugLog) (*action.Con
...
+		config.WithDiscoveryBurst(r.discoveryBurst)
+	}
+	if r.discoveryQPS != -1 {
```
Comment (shlomitk1) [inline pkg/helm/helm.go:314]: I think it's better to refer to positive values only: **> 0**

File: pkg/environment/environment.go (hunk 1 above)
Comment (roytman) [reply pkg/environment/environment.go]: thanks, fixed

File: pkg/helm/helm.go (hunk 2 above)
Comment (roytman) [reply pkg/helm/helm.go:311]: the `GetDiscoveryBurst` and `GetDiscoveryQPS` are defined that they return `-1` if something wrong, so I think checking `-1` is appropriate here.

File: pkg/helm/helm.go (hunk 3 above)
Comment (roytman) [reply pkg/helm/helm.go:314]: as above

Comment (shlomitk1) [reply pkg/helm/helm.go:314]: Still, it is a floating point variable here, using ">" is safer.

Comment (roytman) [reply pkg/helm/helm.go:314]: I don't agree with you, there is the positivity check in the function.

Comment (shlomitk1) [reply pkg/helm/helm.go:314]: Never mind, you are not doing any calculations to be worried about precision.

**Review by shlomitk1 — APPROVED** [review]

**Pull request merged**